scripts/SREP/update-upgrade-policy-state/script.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+# The managed script update_upgrade_policy provides the ability to update the state
+# of the current OpenShift cluster's upgrade policy. It fetches the current state and
+# compares it to the desired state provided, updating it if necessary.
+
+# Usage: 
+# ./script.sh DESIRED_STATE=scheduled 
+# `scheduled` is the desired state you want to set for the cluster's upgrade policy.
+# 
+# Possible values of the upgrade policy state can include "started", "scheduled", "cancelled" etc. 
+# Ensure that you provide a state recognized by your OpenShift Cluster Manager instance.
+
+# Defined Functions:
+
+# Fetch the unique identifier (UUID) of the current OpenShift cluster.
+get_cluster_uuid() {
+    oc get clusterversion version -o json | jq -r '.spec.clusterID'
+}
+
+# Using the cluster's UUID, fetch the cluster's ID.
+get_cluster_id() {
+    ocm list clusters --parameter search="external_id is '${1}'" --columns id --padding 20 --no-headers
+}
+
+# Get upgrade policies associated with the specified cluster ID.
+get_policy_response() {
+    ocm get "/api/clusters_mgmt/v1/clusters/${1}/upgrade_policies/"
+}
+
+# Count the number of upgrade policies available for the current cluster.
+get_policy_count() {
+    echo "$1" | jq '.size'
+}
+
+# Fetch the UUID of the first available upgrade policy for the cluster.
+get_policy_uuid() {
+    echo "$1" | jq -r '.items[0].id'
+}
+
+# Fetch the current state of the specified upgrade policy.
+get_current_state() {
+    ocm get "$1" | jq -r '.value'
+}
+
+# Update the state of the specified upgrade policy to the desired state.
+update_upgrade_policy() {
+    local ocm_slug="$1"
+    local desired_state="$2"
+    local payload="{ \"value\": \"${desired_state}\", \"description\": \"Manually updated by SRE\" }"
+    ocm patch "${ocm_slug}" --body "${payload}" --type application/json
+}
+
+# Parse the arguments passed to the script to determine the desired state.
+parse_arguments() {
+    for arg in "$@"; do
+        case "$arg" in
+            DESIRED_STATE=started)
+                DESIRED_STATE="started"
+                ;;
+            DESIRED_STATE=scheduled)
+                DESIRED_STATE="scheduled"
+                ;;
```

Comment:
I'm interested in how we landed on these three states as the ones an SRE can switch to. I don't think there should be any case where SRE should be changing to a `scheduled` state, as that is controlled wholly by OCM. I'm similarly struggling to think of cases where an SRE would/should be moving the state into `started`, are there any?

`completed`, `failed` and `cancelled` are probably the ones I can think are valid transitions we should allow an SRE to manually move to.

Reply:
@mrbarge Do we even need the status `completed` or `failed` here? I think we only give the ability to set `cancelled` should be sufficient.

Reply:
I'd agree I don't really see much of a use case for `failed`.. however I think `completed` is valid if the cluster has actually upgraded to the version that the upgrade policy says it needs to be on, but MUO somehow isn't doing it.

Worth noting that I'm in the middle of fixing a lot of the need to do this type of handling in MUO with OSD-17840, which is currently in a PR/test phase. But the script is still valuable to have just in case.

Reply:
That sounds fine to me.

Reply:
@mrbarge @bmeng - Thanks a lot for your review feedback. 
Sorry, I wasn't quite clear about what are the valid desired state we should set in the first place and double checked with Bo. I will implement only with the `cancelled` state at this stage.